homeassistant/util/temperature.py

```diff
@@ -9,3 +10,17 @@ def fahrenheit_to_celsius(fahrenheit: float) -> float:
 def celsius_to_fahrenheit(celsius: float) -> float:
     """Convert a Celsius temperature to Fahrenheit."""
     return celsius * 1.8 + 32.0
+
+
+def convert(temperature: float, from_unit: str, to_unit: str) -> (float, str):
+    """Convert a temperature from one unit to another."""
+    if from_unit not in [TEMP_CELSIUS, TEMP_FAHRENHEIT] or \
+        to_unit not in [TEMP_CELSIUS, TEMP_FAHRENHEIT] or \
+            to_unit == from_unit:
+        # If no need or unknown conversion, return temperature
+        return temperature, from_unit
+
+    return (
+        celsius_to_fahrenheit(
+            temperature) if from_unit == TEMP_CELSIUS else round(
```

Comment:
I am really not a big fan of this style, os confusing to read this code. No wonder you hate the linter as this code will have you go > 80 chars all the time.